tests/cpp/multidevice.h

```diff
@@ -45,6 +45,12 @@ class MultiDeviceTest : public NVFuserTest {
       TensorView* tv,
       DeviceIdxType deviceId);
 
+  static at::Tensor shardTensor(
+      at::Tensor tensor,
+      int64_t axis,
+      const DeviceMesh& mesh,
+      DeviceIdxType deviceId);
```

Comment:
I don't mind if you make this method non-static and get deviceId by itself. It saves some typing, and we only use shardTensor within the test which has communicator_ available.

Reply:
Yeah, I think that's a good idea! Makes the tests cleaner